Hakawai/Mentions/HKWMentionsStartDetectionStateMachine.m

```diff
@@ -202,22 +191,36 @@ - (void)characterTyped:(unichar)c asInsertedCharacter:(BOOL)inserted previousCha
     }
 }
 
-- (void)deleteTypedCharacter:(unichar)deletedChar withCharacterNowPrecedingCursor:(unichar)precedingChar {
-    // Determine the type of the character
-    enum CharacterType {
-        CharacterTypeSeparator = 0,
-        CharacterTypeNormal
-    };
-    enum CharacterType deletedCharacterType = ([HKWMentionsStartDetectionStateMachine.separatorSet characterIsMember:deletedChar] || deletedChar == 0)
-    ? CharacterTypeSeparator
-    : CharacterTypeNormal;
-    enum CharacterType currentCharacterType = ([HKWMentionsStartDetectionStateMachine.separatorSet characterIsMember:precedingChar] || precedingChar == 0)
-    ? CharacterTypeSeparator
-    : CharacterTypeNormal;
+- (void)deleteTypedCharacter:(unichar)deletedChar
+withCharacterNowPrecedingCursor:(unichar)precedingChar
+                    location:(NSUInteger)location
+                textViewText:(nonnull NSString *)textViewText {
+    // Determine the character types
+    enum CharacterType deletedCharacterType = [self characterTypeOfCharacter:deletedChar];
+    enum CharacterType currentCharacterType = [self characterTypeOfCharacter:precedingChar];
+
+    __strong __auto_type delegate = self.delegate;
 
     switch (self.state) {
         case HKWMentionsStartDetectionStateQuiescentReady: {
-            if (currentCharacterType == CharacterTypeNormal) {
+            BOOL shouldCreateMention = NO;
+            if (location > 1 && textViewText.length > location - 2) {
+                const unichar characterBeforePrecedingChar = [textViewText characterAtIndex:location-2];
+                shouldCreateMention = [HKWMentionsStartDetectionStateMachine.separatorSet characterIsMember:characterBeforePrecedingChar];
+            }
+            // If user deletes white-space between control character and word, then query mention with word next to whitepace.
+            if (currentCharacterType == CharacterTypeControlCharacter
+                && (deletedCharacterType == CharacterTypeSeparator || deletedCharacterType == CharacterTypeControlCharacter)
```

Comment:
isn't the controlCharacter part of this logic handled by "secondPreviousCaracterIsControl" logic above?

Reply:
We were only checking for `characterBeforePrecedingChar`'s type to set `shouldCreateMention`.  We have now included `precedingCharacterType == CharacterTypeControlCharacter` check in `canCreateMention` (previously `shouldCreateMention`) and removed the same check from here.